<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>666</width>
    <height>610</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset theme="rotate.png">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <property name="windowFilePath">
   <string/>
  </property>
  <property name="animated">
   <bool>true</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="cartStatus">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>161</width>
      <height>16</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: lightgreen</string>
    </property>
    <property name="text">
     <string>Cart Status</string>
    </property>
   </widget>
   <widget class="QLabel" name="cartStatus_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Cart Command:</string>
    </property>
   </widget>
   <widget class="QLabel" name="currentCommand">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>50</y>
      <width>81</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>STOP</string>
    </property>
   </widget>
   <widget class="QLabel" name="cartStatus_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>100</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Distance to Obstacle:</string>
    </property>
   </widget>
   <widget class="QLabel" name="cartStatus_4">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Distance to Abyss:</string>
    </property>
   </widget>
   <widget class="QLabel" name="obstacleHeadcam">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>100</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="abyssHeadcam">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>120</y>
      <width>41</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="cartStatus_5">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>142</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="cartStatus_6">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>142</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="cartStatus_7">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>142</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Yaw</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="cartStatus_8">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>160</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Current</string>
    </property>
   </widget>
   <widget class="QLabel" name="cartStatus_9">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>180</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Target</string>
    </property>
   </widget>
   <widget class="QLabel" name="currentX">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>160</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="currentY">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>160</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="currentYaw">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>160</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="targetX">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>180</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="targetY">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>180</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="targetYaw">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>180</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>261</width>
      <height>211</height>
     </rect>
    </property>
    <property name="title">
     <string>State</string>
    </property>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>80</y>
       <width>44</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Headcam</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>80</y>
       <width>38</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Sensors</string>
     </property>
    </widget>
    <widget class="QLabel" name="obstacleSensors">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>100</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="abyssSensors">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>120</y>
       <width>41</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>-</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupRotate">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>220</y>
      <width>201</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Rotate Cart</string>
    </property>
    <widget class="QSpinBox" name="rotationDegrees">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>30</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
    </widget>
    <widget class="QLabel" name="cartStatus_12">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotate by</string>
     </property>
    </widget>
    <widget class="QLabel" name="cartStatus_11">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>30</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>degrees</string>
     </property>
    </widget>
    <widget class="QLabel" name="rotateLeft">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>70</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>anticlockRotation.jpg</pixmap>
     </property>
    </widget>
    <widget class="QLabel" name="rotateRight">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>70</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>clockRotation.jpg</pixmap>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupMove">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>201</width>
      <height>251</height>
     </rect>
    </property>
    <property name="title">
     <string>Move Cart</string>
    </property>
    <widget class="QLabel" name="moveRight">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>120</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>right.JPG</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="moveForward">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>60</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>forward.JPG</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="moveDiagonalRightBackward">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>160</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>diagonalRightBackward.JPG</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QSpinBox" name="moveDistanceCm">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>120</y>
       <width>61</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="moveLeft">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>left.JPG</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="moveDiagonalLeftForward">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>diagonalLeftForward.JPG</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="moveDiagonalRightForward">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>80</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>diagonalRightForward.JPG</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="cartStatus_22">
     <property name="geometry">
      <rect>
       <x>84</x>
       <y>100</y>
       <width>26</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Move</string>
     </property>
    </widget>
    <widget class="QLabel" name="moveBackward">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>180</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>backward.JPG</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="moveDiagonalLeftBackward">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>160</y>
       <width>28</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap>diagonalLeftBackward.JPG</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="cartStatus_23">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>143</y>
       <width>13</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>cm</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>30</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Speed</string>
     </property>
    </widget>
    <widget class="QSlider" name="cartSpeed">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>30</y>
       <width>121</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>80</number>
     </property>
     <property name="maximum">
      <number>230</number>
     </property>
     <property name="value">
      <number>180</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>220</y>
       <width>30</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Done:</string>
     </property>
    </widget>
    <widget class="QSlider" name="moveProgress">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>220</y>
       <width>121</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>300</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="sliderPosition">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="sensorArea">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>100</y>
      <width>231</width>
      <height>341</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
   </widget>
   <zorder>groupBox</zorder>
   <zorder>cartStatus</zorder>
   <zorder>cartStatus_2</zorder>
   <zorder>currentCommand</zorder>
   <zorder>cartStatus_3</zorder>
   <zorder>cartStatus_4</zorder>
   <zorder>obstacleHeadcam</zorder>
   <zorder>abyssHeadcam</zorder>
   <zorder>cartStatus_5</zorder>
   <zorder>cartStatus_6</zorder>
   <zorder>cartStatus_7</zorder>
   <zorder>cartStatus_8</zorder>
   <zorder>cartStatus_9</zorder>
   <zorder>currentX</zorder>
   <zorder>currentY</zorder>
   <zorder>currentYaw</zorder>
   <zorder>targetX</zorder>
   <zorder>targetY</zorder>
   <zorder>targetYaw</zorder>
   <zorder>groupRotate</zorder>
   <zorder>groupMove</zorder>
   <zorder>sensorArea</zorder>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
